src/HealthGPS/analysis_module.cpp

```diff
@@ -66,7 +101,13 @@ void AnalysisModule::initialise_population(RuntimeContext &context) {
     publish_result_message(context);
 }
 
-void AnalysisModule::update_population(RuntimeContext &context) { publish_result_message(context); }
+void AnalysisModule::update_population(RuntimeContext &context) {
+
+    // Reset the calculated factors vector to 0.0
+    std::fill(calculated_factors_.begin(), calculated_factors_.end(), 0.0);
```

Comment:
Might be able to use `std::ranges::fill(calculated_factors_, 0.0)` here, but it *might* be C++26. Also insertion defaults to `0.0` anyway, so value is not needed.

Reply:
I'm assuming this is placeholder code and that something else will follow resetting the elements to zero? Fine for now, but either way @jamesturner246 is right that eventually you might not need to explicitly zero the vector